src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs

```diff
@@ -720,9 +720,24 @@ private record RequestState(
         public int ActiveConcurrentConnections { get; set; }
     };
 
-    private class TestSystemClock : ISystemClock
+    private class TestTimeProvider : TimeProvider
```

Comment:
Should it override `CreateTimer` throwing `NotImplementedException` to ensure that no current or future code is using it as it is not mocked?

Reply:
This would be called indirectly by `Task.WaitAsync` or `Task.DelayAsync` so it could be missed.